src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -541,11 +533,31 @@ private bool TryParseBindAttribute(BindEntry bindEntry, out string valueAttribut
             // We expect 1 bind tag helper per-node.
             var node = bindEntry.BindNode;
             var attributeName = node.AttributeName;
+
+            // Even though some of our 'bind' tag helpers specify the attribute names, they
+            // should still satisfy one of the valid syntaxes.
+            if (!TryParseBindAttribute(bindEntry, out valueAttributeName))
+            {
+                return false;
+            }
+
             valueAttributeName = node.TagHelper.GetValueAttributeName() ?? valueAttributeName;
 
             // If there an attribute that specifies the event like @bind:event="oninput",
             // that should be preferred. Otherwise, use the one from the tag helper.
-            changeAttributeName ??= node.TagHelper.GetChangeAttributeName();
+            if (bindEntry.BindEventNode == null)
+            {
+                changeAttributeName ??= node.TagHelper.GetChangeAttributeName();
+            }
+            else if (TryExtractEventNodeStaticText(bindEntry.BindEventNode, out var text))
+            {
+                // If we have a node for @bind:event - we may or may not be able to know the attribute name.
+                changeAttributeName = text;
+            }
+            else
+            {
+                changeAttributeNode = ExtractEventNodeExpression(bindEntry.BindEventNode);
+            }
```

Comment:
Can you give an example for when each of these conditions will be true?

Reply:
sure.